src/service/users.js

```diff
@@ -237,3 +237,20 @@ export const checkFollowStatusBatch = async (targetUserIds = [], axiosOptions =
     );
   }
 };
+
+export const checkFollowStatus = async (channelId) => {
+  if (!channelId) return Promise.reject(new Error('Target User ID is empty'));
+
+  try {
+    const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+    if (result?.status === 200) {
+      return Promise.resolve(result?.data?.data);
+    }
+
+    return Promise.reject(result?.data?.message);
+  } catch (e) {
+    return Promise.reject(
+      e?.response?.data?.message || "Can't check follow status, please try again later"
+    );
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkFollowStatus` function seems to be well implemented. It checks if the channelId is provided, makes an API call, and handles potential errors. However, it's a good practice to encode URI components to ensure that special characters are correctly passed to the server. In this case, the `channelId` should be encoded.

```diff
- const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+ const result = await api.get(`/users/check-follow-by-channelId?channelId=${encodeURIComponent(channelId)}`);
```
Also, consider using HTTP status codes constants for better readability and maintainability of your code.

```diff
- if (result?.status === 200) {
+ if (result?.status === HttpStatus.OK) {
```
You would need to import the `HttpStatus` enum from a library like `http-status-codes` or define it yourself.


<!-- This is an auto-generated comment by OSS CodeRabbit -->